src/lambda/wizards/samInitWizard.ts

```diff
@@ -1,552 +1,257 @@
+        buttons: makeButtons(samInitDocUrl),
+    })
+}
 
-            const choices = await picker.promptUser({
-                picker: quickPick,
-                onDidTriggerButton: (button, resolve, reject) => {
-                    if (button === vscode.QuickInputButtons.Back) {
-                        resolve(undefined)
-                    } else if (button === this.helpButton) {
-                        vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
-                    }
-                },
-            })
-            const val = picker.verifySinglePickerOutput(choices)
+function createTemplatePrompter(
+    currRuntime: Runtime,
+    packageType: RuntimePackageType,
+    samCliVersion: string
+): QuickPickPrompter<SamTemplate> {
+    const templates = getSamTemplateWizardOption(currRuntime, packageType, samCliVersion)
+    const items = templates.toArray().map(template => ({
+        label: template,
+        data: template,
+        detail: getTemplateDescription(template),
+    }))
+
+    return createQuickPick(items, {
+        title: localize('AWS.samcli.initWizard.template.prompt', 'Select a SAM Application Template'),
+        buttons: makeButtons(samInitDocUrl),
+    })
+}
 
-            return val ? (val.label as DependencyManager) : undefined
-        }
-    }
+function createSchemaRegionPrompter(context: {
+    regionProvider: RegionProvider
+    awsContext: AwsContext
+}): QuickPickPrompter<string> {
+    return createRegionPrompter(context, {
+        title: localize('AWS.samcli.initWizard.schemas.region.prompt', 'Select an EventBridge Schemas Region'),
+        buttons: makeButtons(eventBridgeSchemasDocUrl),
+        serviceId: 'schemas',
+    }).transform(r => r.id)
+}
 
-    public async promptUserForTemplate(
-        currRuntime: Runtime,
-        packageType: RuntimePackageType,
-        currTemplate?: SamTemplate
-    ): Promise<SamTemplate | undefined> {
-        this.stepsToAdd.promptUserForRegion = false
-        const templates = getSamTemplateWizardOption(currRuntime, packageType, this.samCliVersion)
-        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
-            options: {
-                ignoreFocusOut: true,
-                title: localize('AWS.samcli.initWizard.template.prompt', 'Select a SAM Application Template'),
-                value: currTemplate,
-                step: 2 + this.additionalSteps(),
-                totalSteps: this.totalSteps + this.additionalSteps(),
-            },
-            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
-            items: templates.toArray().map(template => ({
-                label: template,
-                alwaysShow: template === currTemplate,
-                detail:
-                    template === currTemplate
-                        ? localize('AWS.wizard.selectedPreviously', 'Selected Previously')
```

Comment:
The new code removes the 'Selected Previously' stuff because the last selection is automatically highlighted now when going back.

Reply:
Added as an option that's applied by default to all prompts